terraso_backend/apps/export/views.py

```diff
@@ -0,0 +1,306 @@
+# Copyright © 2021-2025 Technology Matters
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see https://www.gnu.org/licenses/.
+
+from urllib.parse import unquote
+
+from django.contrib.auth import get_user_model
+from django.http import HttpResponse, HttpResponseBadRequest, JsonResponse
+
+from .fetch_data import fetch_site_data, fetch_soil_id
+from .fetch_lists import fetch_all_sites, fetch_project_list, fetch_user_owned_sites
+from .formatters import sites_to_csv
+from .html_pages import export_page_html, invalid_token_page
+from .models import ExportToken
+from .transformers import transform_site_data
+
+
+def _resolve_export_token(token):
+    """
+    Resolve export token to (resource_type, resource_id).
+    Returns invalid token page if token not found.
+    """
+    try:
+        export_token = ExportToken.objects.get(token=token)
+        return export_token.resource_type, export_token.resource_id
+    except ExportToken.DoesNotExist:
+        return None, None
+
+
+def _setup_system_export_user(request):
+    """Set up request with system export user and flag."""
+    User = get_user_model()
+    try:
+        service_user = User.objects.get(email="system-export@terraso.org")
+    except User.DoesNotExist:
+        raise RuntimeError(
+            "System export user not found. Run migrations to create 'system-export@terraso.org' user."
+        )
+    request.user = service_user
+    request.is_system_export = True
```

Comment:
still trying to wrap my head around why we need the system export user: it seems that a user should not be able to create an export token for resources to which they do not have access. so if the resources associated with the export token are accessible by the user, would we be able to set the request's user to the user who owns the export token, and then rely on our existing authentication logic rather than special-casing with the `system-export` user? or is there some reason i'm not thinking of why this wouldn't work?

Reply:
i guess my opinion would be that it's cleaner & more aligned with the actual intention to be doing something more like an assume role here, since otherwise we're kind of duplicating the permissions logic (which is quite simple in this case, so not a big deal. but i also see how that could seem riskier in other ways, since the system export user only gets to bypass permissions checks in specific blessed ways, whereas an assume role could create other holes.

maybe we can have a quick convo about this? it does seem like it'd simplify the code